lib/src/net.dart

```diff
@@ -1,373 +1,352 @@
-            Map optargs = {};
-            _global_optargs.forEach((k,v){
-              optargs[k] = v is RqlQuery ? v.build() : v;
-            });
+class Query extends RqlQuery {
+  p.Query_QueryType _type;
+  int _token;
+  RqlQuery _term;
+  Map _global_optargs;
+  Cursor _cursor;
+  final Completer _queryCompleter = new Completer();
+
+  Query(p.Query_QueryType this._type, int this._token, [RqlQuery this._term, Map this._global_optargs]);
+
+  serialize() {
+    List res = [_type.value];
+    if (_term != null) {
+      res.add(_term.build());
+    }
+    if (_global_optargs != null) {
+      Map optargs = {};
+      _global_optargs.forEach((k, v) {
+        optargs[k] = v is RqlQuery ? v.build() : v;
+      });
 
-            res.add(optargs);
-        }
-        return JSON.encode(res);
+      res.add(optargs);
     }
+    return JSON.encode(res);
+  }
 }
 
-class Response{
-    int _token;
-    int _type;
-    var _data;
-    var _backtrace;
-    var _profile;
-    List _notes = [];
-    
-    Response(int this._token,String json_str){
-        if(json_str.length > 0){
-        Map full_response = JSON.decode(json_str);
-        this._type = full_response["t"];
-        this._data = full_response["r"];
-        this._backtrace = full_response["b"];
-        this._profile = full_response["p"];
-        this._notes = full_response["n"];
-        }
+class Response {
+  int _token;
+  int _type;
+  var _data;
+  var _backtrace;
+  var _profile;
+  List _notes = [];
+
+  Response(int this._token, String json_str) {
+    if (json_str.length > 0) {
+      Map full_response = JSON.decode(json_str);
+      this._type = full_response["t"];
+      this._data = full_response["r"];
+      this._backtrace = full_response["b"];
+      this._profile = full_response["p"];
+      this._notes = full_response["n"];
     }
+  }
 }
 
 class Connection {
+  var _socket = null;
+  int _next_token = 1;
+  String _host;
+  int _port;
+  String _db;
+  String _auth_key;
+  Map _ssl;
 
-    Socket _socket = null;
-    int _next_token = 1;
-    String _host;
-    int _port;
-    String _db;
-    String _auth_key;
-
-    Completer _completer = new Completer();
-
-    int _response_length = 0;
-    List<int> _response_buffer = [];
-
-    final Map _replyQueries = new Map();
-    final Queue<Query> _sendQueue = new Queue<Query>();
+  Completer _completer = new Completer();
 
-    final Map <String, List> _listeners = new Map<String, List>();
+  int _response_length = 0;
+  List<int> _response_buffer = [];
 
+  final Map _replyQueries = new Map();
+  final Queue<Query> _sendQueue = new Queue<Query>();
 
-    Connection(String this._db, String this._host, int this._port, String this._auth_key);
+  final Map<String, List> _listeners = new Map<String, List>();
 
-    get isClosed => _socket == null;
-    
-    use(String db){
-        _db = db;
+  Connection(String this._db, String this._host, int this._port, String this._auth_key, {Map ssl}) {
```

Comment:
Change constructor to take optional SSL argument. Follows the same api as the official driver